src/coreclr/src/vm/arm64/asmhelpers.S

```diff
@@ -1262,6 +1262,26 @@ GenerateProfileHelper ProfileTailcall, PROFILE_TAILCALL
 
 #endif
 
+#define PAGE_SIZE_LOG12 12
+#define PAGE_SIZE 4096
+
+LEAF_ENTRY JIT_StackProbe, _TEXT
```

Comment:
I think you should add a header comment with "on entry" and "on exit" conditions spelled out.